qiskit/tools/visualization/_matplotlib.py

```diff
@@ -480,7 +527,7 @@ def _draw_ops(self, verbose=False):
             layer_width = 1
 
             for op in layer:
-                if op.name in _wide_gate:
+                if op.name in _wide_gate or op.name not in self._style.dispcol:
```

Comment:
I think that users should be able to color their custom gates by adding them to `_style.dispcol`, so maybe it would be better to check if the gate exists in some list of predefined gates?

Reply:
That's something I struggled with when writing this patch, there really isn't a good list to check against whether the gate is custom or not. I'd have to hard code a list of all the standard gate names in the module to accomplish this, which didn't seem great. 

But, this check is only needed for outlining how wide the gate is and since I defaulted to a wide gate for customs. It probably would be better to just determine this dynamically based on the length of the actual op.name and set the `wide` parameter dynamically based on that. I left that as a future improvement in the commit message, but maybe I should stop being lazy and just add it. :) I'll work on that in a bit.

Reply:
I agree hard coding it isn't ideal, but all the standard gate names are hardcoded anyway throughout the code so I don't think having an overall list of them is too bad, and might be a good solution in the mean time before we dynamically size the gates.

Reply:
I just implemented the dynamic spacing and sizing for gates to avoid this with: https://github.com/Qiskit/qiskit-terra/pull/2070/commits/e9fa8afebb4af848fd4bd2bae947298817500f3f